### ISSUE TITLE
Unable to Build Quickjournal Example in Release Mode because of Linker Error

### ISSUE THREAD

**Issue by escamoteur** [realm/realm-dotnet#606]
Hi,

I just tried to build your QuickJournal in Release Mode and git an Linker Error.
As this is my very first encounter with Realm, it's pretty dissapointing.

```
1>------ Build started: Project: QuickJournal.Droid, Configuration: Release Any CPU ------
1>    Fody: Fody (version 1.29.4.0) Executing
1>    Fody:   Finished Fody 154ms.
1>  QuickJournal.Droid -> C:\Entwicklung\Tests\realm-dotnet\examples\QuickJournal\Droid\bin\Release\QuickJournal.Droid.dll
1>    Fody:   Skipped Verifying assembly since it is disabled in configuration
1>    Fody:   Finished verification in 0ms.
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018: The "LinkAssemblies" task failed unexpectedly.
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018: Xamarin.Android.XamarinAndroidException: error XA2006: Reference to metadata item 'System.String System.String::Format(System.IFormatProvider,System.String,System.Object,System.Object)' (defined in 'Realm, Version=0.75.0.0, Culture=neutral, PublicKeyToken=null') from 'Realm, Version=0.75.0.0, Culture=neutral, PublicKeyToken=null' could not be resolved. ---> Mono.Cecil.ResolutionException: Failed to resolve System.String System.String::Format(System.IFormatProvider,System.String,System.Object,System.Object)
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Mono.Linker.Steps.MarkStep.MarkMethod(MethodReference reference)
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Mono.Linker.Steps.MarkStep.MarkInstruction(Instruction instruction)
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Mono.Linker.Steps.MarkStep.MarkMethodBody(MethodBody body)
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Mono.Linker.Steps.MarkStep.ProcessMethod(MethodDefinition method)
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Mono.Linker.Steps.MarkStep.ProcessQueue()
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Mono.Linker.Steps.MarkStep.Process()
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Mono.Linker.Steps.MarkStep.Process(LinkContext context)
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Mono.Linker.Pipeline.Process(LinkContext context)
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at MonoDroid.Tuner.Linker.Process(LinkerOptions options, LinkContext& context)
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Xamarin.Android.Tasks.LinkAssemblies.Execute()
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    --- End of inner exception stack trace ---
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Xamarin.Android.Diagnostic.Error(Int32 code, Exception innerException, String message, Object[] args)
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Xamarin.Android.Tasks.LinkAssemblies.Execute()
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
1>C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1473,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
2>------ Skipped Deploy: Project: QuickJournal.Droid, Configuration: Release Any CPU ------
2>Project not selected to build for this solution configuration 
========== Build: 0 succeeded, 1 failed, 1 up-to-date, 0 skipped ==========
========== Deploy: 0 succeeded, 0 failed, 1 skipped ==========
```

#### Version of Realm and tooling
Realm version(s): 0.75

Xamarin/Visual Studio version: ?
VS2015 V2

Which operating system version and device: ?
Windows 10

Android 5.1

**Comment by Rohitwagh** [comment]
I am facing the same issue with linker set to SDK assemblies while trying to build my app on Android Release mode:

>Error	40	The "LinkAssemblies" task failed unexpectedly.
Xamarin.Android.XamarinAndroidException: error XA2006: Reference to metadata item 'System.String System.String::Format(System.IFormatProvider,System.String,System.Object,System.Object)' (defined in 'Realm, Version=0.75.0.0, Culture=neutral, PublicKeyToken=null') from 'Realm, Version=0.75.0.0, Culture=neutral, PublicKeyToken=null' could not be resolved. ---> Mono.Cecil.ResolutionException: Failed to resolve System.String System.String::Format(System.IFormatProvider,System.String,System.Object,System.Object)
   at Mono.Linker.Steps.MarkStep.MarkMethod(MethodReference reference)
   at Mono.Linker.Steps.MarkStep.MarkInstruction(Instruction instruction)
   at Mono.Linker.Steps.MarkStep.MarkMethodBody(MethodBody body)
   at Mono.Linker.Steps.MarkStep.ProcessMethod(MethodDefinition method)
   at Mono.Linker.Steps.MarkStep.ProcessQueue()
   at Mono.Linker.Steps.MarkStep.Process()
   at Mono.Linker.Steps.MarkStep.Process(LinkContext context)
   at Mono.Linker.Pipeline.Process(LinkContext context)
   at MonoDroid.Tuner.Linker.Process(LinkerOptions options, LinkContext& context)
   at Xamarin.Android.Tasks.LinkAssemblies.Execute()
   --- End of inner exception stack trace ---
   at Xamarin.Android.Diagnostic.Error(Int32 code, Exception innerException, String message, Object[] args)
   at Xamarin.Android.Tasks.LinkAssemblies.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__20.MoveNext()	MobileApp.Droid

Realm: 0.75.0
Xamarin   4.0.3.214 (0dd817c)
Xamarin.Android   6.0.3.5 (a94a03b)
VS 2013
Android version 6.0 - API 23
Samsung EDGE 6 - SM G9251

**Comment by kristiandupont** [comment]
This was just resolved with #591. There will be a release with this fix very soon, probably this week.
